gtsfm/densify/mvs_base.py

```diff
@@ -0,0 +1,62 @@
+"""Base class for the MVS stage of the back-end.
+
+Authors: John Lambert, Ren Liu
+"""
+import abc
+from typing import Dict, Tuple
+
+import dask
+import numpy as np
+from dask.delayed import Delayed
+from gtsam import PinholeCameraCal3Bundler
+
+from gtsfm.common.image import Image
+
+
+class MvsBase(metaclass=abc.ABCMeta):
+    """Base class for all multi-view stereo hyperparameters."""
+
+    def __init__(self) -> None:
+        """Initialize the MVS module """
+        pass
+
+    @abc.abstractmethod
+    def densify(
+        self,
+        images: Dict[int,Image],
+        cameras: Dict[int, PinholeCameraCal3Bundler],
+        min_distance: float,
+        max_distance: float,
+    ) -> Tuple[np.ndarray, Dict[int, np.ndarray]]:
+        """Densify a point cloud using multi-view stereo
+
+        Args:
+            image: dictionary mapping image indices to input images.
+            cameras: dictionary mapping image indices to camera parameters
+            min_distance: minimum distance from any camera to any 3d point
+            max_distance: maximum distance from any camera to any 3d point
+
+        Returns:
+            array of shape (N,3) representing dense point cloud
+            dictionary mapping integer index to depth map of shape (H,W)
```

Comment:
do we need the depth map?

Reply:
I do not get what this return type is. Depth map for each image?

Maybe the docstring can be `Depth map for each input image`.

Reply:
That's right -- depth map for each image. @codyly do you think it is useful?

Technically, to calculate MVS completeness metrics, we need to align both point clouds, and project them into the same space. I prefer we align the estimate to the ground truth. So that means unless we pass in the ground truth here, we would need to re-compute the depth maps again afterwards. So maybe returning the depth maps as well here is not helpful.